scripts/file-entities.php

```diff
@@ -365,3 +309,23 @@ function writeEntity( $file , Entity $ent )
 
     fwrite( $file , $line );
 }
+
+function realpain( string $path , bool $touch = false ) : string
+{
+    // pain is real
+
+    // care for external XML tools (realpath everywhere)
+    // care for Windows builds (foward slashes everywhere)
+    // avoid `cd` and chdir() like the plague
```

Comment:
This seems only relevant for ZTS builds (which are not Windows specific); maybe we should require NTS builds?

Reply:
I do not understand what ZTS/NTS will change in this context (this code is used in command line). Avoiding `cd` and `chdir()` in all places means that two different code paths, included or exec(), will not be affected by cd/chdir that got executed only sometimes.

Reply:
For NTS builds, `chdir()` acutally changes the CWD of the process, so that `exec()` will start in that folder; for ZTS that isn't the case ("virtual directory").

Anyhow, using absolute paths everywhere works uniformly.

Reply:
ZTS then will fix the issue for `exec()` ed code, but not for included code.